### ISSUE TITLE
The decimal type is not serialized

### ISSUE THREAD

**Issue by tsadowski** [xoofx/Tomlyn#62]
There is no toString method in TomlFormatHelper defined for the decimal floating point type. Decimal numbers are then simply missing from the output. 

Reading numbers into a decimal type works.

Cheers,
Torsten

**Comment by lilith** [comment]
Toml (the spec) supports only binary64/double/f64 representable numbers. There's overlap between the numbers representable in binary64 and decimal, but both can express numbers that the other can't. This makes for a particularly challenging situation to implement, unless we were to use a 29 digit string literal or something. I prototyped a solution that worked until the unit tests started testing decimal.MaxValue. It can't roundtrip, unfortunately.

**Comment by lilith** [comment]
Here's my branch that explored this: https://github.com/lilith/Tomlyn/tree/decimal-support
Honestly the safest approach might be to explicitly document decimal is not supported, and also throw an exception on use. 

Reading numbers into a decimal type won't work if the double is too small or large.

**Comment by xoofx** [comment]
> Honestly the safest approach might be to explicitly document decimal is not supported, and also throw an exception on use.

Yep, agreed. TOML was not meant for portable data but for easy configuration.
